`make` does not verify that xorriso exists initially

**Reporter info**

```
 * Distribution - (run ```lsb_release -a```)
aaron@toph:~/Documents/Code/pop-distro$ lsb_release -a
LSB Version:	core-9.20160110ubuntu5-amd64:core-9.20160110ubuntu5-noarch:printing-9.20160110ubuntu5-amd64:printing-9.20160110ubuntu5-noarch:security-9.20160110ubuntu5-amd64:security-9.20160110ubuntu5-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty

 * Related Application and/or Package Version - apt policy $PACKAGE NAME
aaron@toph:~/Documents/Code/pop-distro$ apt policy xorriso
xorriso:
  Installed: 1.4.6-1
  Candidate: 1.4.6-1
  Version table:
 *** 1.4.6-1 500
        500 http://us.archive.ubuntu.com/ubuntu zesty/universe amd64 Packages
        100 /var/lib/dpkg/status

```

**Issue/Bug Description**
I attempted to `make` and it errored out because I did not already have `xorriso` installed.

```
aaron@toph:~/Documents/Code/pop-distro$ make
mkdir -p build/17.04
wget -O "build/17.04/ubuntu.iso" "http://cdimage.ubuntu.com/ubuntu-gnome/releases/17.04/release/ubuntu-gnome-17.04-desktop-amd64.iso"
--2017-07-29 20:15:55--  http://cdimage.ubuntu.com/ubuntu-gnome/releases/17.04/release/ubuntu-gnome-17.04-desktop-amd64.iso
Resolving cdimage.ubuntu.com (cdimage.ubuntu.com)... 91.189.88.165
Connecting to cdimage.ubuntu.com (cdimage.ubuntu.com)|91.189.88.165|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1517780992 (1.4G) [application/x-iso9660-image]
Saving to: ‘build/17.04/ubuntu.iso’

build/17.04/ubuntu.iso 100%[===========================>]   1.41G  4.40MB/s    in 7m 41s  

2017-07-29 20:23:37 (3.14 MB/s) - ‘build/17.04/ubuntu.iso’ saved [1517780992/1517780992]

# Remove old ISO
sudo rm -rf "build/17.04/iso"
[sudo] password for aaron: 
# Extract ISO
xorriso -acl on -xattr on -osirrox on -indev "build/17.04/ubuntu.iso" -extract / "build/17.04/iso"
/bin/sh: 1: xorriso: not found
Makefile:116: recipe for target 'build/17.04/iso_extract.tag' failed
make: *** [build/17.04/iso_extract.tag] Error 127
```


**Steps to reproduce (if you know)**
```
$ sudo apt remove xorriso
$ cd ~/path/to/pop_os
$ make
```

**Expected behaviour**
When executing `make`, it should verify that all the relevant commands exist prior to running. I expect something like:
```
$ make
Dependency error: missing "xorriso"
```

or better yet:
```
$ make
Missing 'xorriso', installing now:
sudo apt install xorriso
[sudo] password for USERNAME:
```

even better if it just lists it as a package dependency for the Makefile. I forget if Makefiles can do that. (it's been a while)

**Other Notes**
My shell-scripting is rusty but I'll attempt to correct this.

--- a/Makefile
+++ b/Makefile
@@ -62,8 +62,22 @@ SED=\
 	s|UBUNTU_CODE|$(UBUNTU_CODE)|g; \
 	s|UBUNTU_NAME|$(UBUNTU_NAME)|g
 
-# Does the xorriso command exist?
 XORRISO=$(shell command -v xorriso 2> /dev/null)
+ZSYNC=$(shell command -v zsync 2> /dev/null)
+SQUASHFS=$(shell command -v mksquashfs 2> /dev/null)
+
+# Ensure that `zsync` is installed already
+ifeq (,$(ZSYNC))
+$(error zsync not found! Run deps.sh first.)
+endif
+# Ensure that `xorriso` is installed already
+ifeq (,$(XORRISO))
+$(error xorriso not found! Run deps.sh first.)
+endif
+# Ensure that `squashfs` is installed already
+ifeq (,$(SQUASHFS))
+$(error squashfs-tools not found! Run deps.sh first.)
+endif
 
 .PHONY: all clean iso qemu qemu_uefi qemu_ubuntu qemu_ubuntu_uefi zsync
 
@@ -116,10 +130,6 @@ zsync: $(BUILD)/ubuntu.iso
 	zsync "$(UBUNTU_ISO).zsync" -o "$<"
 
 $(BUILD)/iso_extract.tag: $(BUILD)/ubuntu.iso
-# Ensure that `xorriso` is installed already
-ifeq (,$(XORRISO))
-	$(error xorriso not found! Run deps.sh first.)
-endif
 
 	# Remove old ISO
 	sudo rm -rf "$(BUILD)/iso"

--- a/Makefile
+++ b/Makefile
@@ -118,8 +118,7 @@ zsync: $(BUILD)/ubuntu.iso
 $(BUILD)/iso_extract.tag: $(BUILD)/ubuntu.iso
 # Ensure that `xorriso` is installed already
 ifeq (,$(XORRISO))
-	@echo "xorriso not found, installing..."
-	sudo apt install -y xorriso
+	$(error xorriso not found! Run deps.sh first.)
 endif
 
 	# Remove old ISO

--- a/Makefile
+++ b/Makefile
@@ -62,6 +62,9 @@ SED=\
 	s|UBUNTU_CODE|$(UBUNTU_CODE)|g; \
 	s|UBUNTU_NAME|$(UBUNTU_NAME)|g
 
+# Does the xorriso command exist?
+XORRISO=$(shell command -v xorriso 2> /dev/null)
+
 .PHONY: all clean iso qemu qemu_uefi qemu_ubuntu qemu_ubuntu_uefi zsync
 
 iso: $(BUILD)/$(DISTRO_CODE).iso
@@ -113,6 +116,12 @@ zsync: $(BUILD)/ubuntu.iso
 	zsync "$(UBUNTU_ISO).zsync" -o "$<"
 
 $(BUILD)/iso_extract.tag: $(BUILD)/ubuntu.iso
+# Ensure that `xorriso` is installed already
+ifeq (,$(XORRISO))
+	@echo "xorriso not found, installing..."
+	sudo apt install -y xorriso
+endif
+
 	# Remove old ISO
 	sudo rm -rf "$(BUILD)/iso"
 

